### ISSUE TITLE
Cirrus: Failed to stop: java something

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#18065]
Seeing this a lot lately, multiple times per day:
```
Failed to stop: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@1b21f19b[
   Not completed, task = java.util.concurrent.Executors$RunnableAdapter@2a087959[
      Wrapped task = TrustedListenableFutureTask@5768a9f6[
         status=PENDING, info=[
            task=[running=[NOT STARTED YET],
            com.google.api.gax.rpc.CheckingAttemptCallable@5137f960]
         ]
      ]
   ]
] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@728a8117[
   Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 4438
]
```
e.g. [here](https://cirrus-ci.com/task/6007844865048576), [here](https://cirrus-ci.com/task/4566393509642240). Unfortunately my flake logger has a bug, it isn't preserving log links for these. I'll track that down when I have time. ITM this is still worth filing.

Obvious first thought is that it's a problem on the Cirrus end, but I see [nothing reported there](https://github.com/cirruslabs/cirrus-ci-web/issues?q=is%3Aissue+is%3Aopen+failed+to+stop).

We're also seeing lots of `Failed to start`, but those aren't as bad, because those happen within a few minutes. The `stop` ones happen after the task has finished, and all tests have run (possibly even passed), so it's a big waste of energy and time.

**Comment by cevich** [comment]
No current orphans on GCP, but I remember we did have a handful on EC2.  Unfortunately there's no easy way to associate EC2 VMs with the task that spawned them.  So this is just speculation.

**Comment by edsantiago** [comment]
"EC2" means "aarch64", doesn't it? I don't see any aarch64 failures in my flake history. (I do have flake history, just, no logs).

**Comment by cevich** [comment]
EC2 is for the aarch64, podman-machine, and windows tests.  IIRC, the orphaned VMs I killed the other day were for all three (one of each).

**Comment by fkorotkov** [comment]
> Unfortunately there's no easy way to associate EC2 VMs with the task that spawned them.

If the IAM Cirrus is using has `ec2:CreateTags` permission, you can enable `experimental` flag on the `ec2_instance` tasks:

```yaml
task:
  experimental: true
  ec2_instance:
    # ...
```

This way you'll see pretty names for the VMs on AWS.

As for the reported two tasks. These are on GCP and seems the error happens after Cirrus successfully requests deletion of an instance so things should be cleanedup. I've added handling of such error so tasks won't get in failed state.

The orphan AWS instances are concerning and I'll appreciate if you can provide some info about the VMs if you have. I might try to find some information in internal logs.  Ideally I'll appreciate if you can add the permissions and enable `experimental` flag so we can catch task ids for such VMs.

**Comment by cevich** [comment]
@fkorotkov thanks.  IIRC, I already setup the EC2 tagging permissions so I think all that's needed is turning on the `experimental` flag.  I'll open a PR to enable that for podman, which has the heaviest usage.

The other thing that occurred to me is: I/we can examine the EC2 instance user-data (script).  I think the task or build ID is passed on the Cirrus-agent command-line, no?

In any case, I'll keep an eye out for these and let you know next time it happens.

**Comment by fkorotkov** [comment]
@cevich good catch! Yes, the user data contains a bootstrap script that has a task id and credentials for reporting updates and logs.

**Comment by cevich** [comment]
Thanks for confirming.  I'll keep my eyes out and let you know.
